src/Microsoft.AspNet.Mvc.Razor.Host/DesignTimeRazorPathNormalizer.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.Razor.Internal
+{
+    public class DesignTimeRazorPathNormalizer : RazorPathNormalizer
+    {
+        private readonly string _applicationRoot;
+
+        public DesignTimeRazorPathNormalizer([NotNull] string applicationRoot)
+        {
+            _applicationRoot = applicationRoot;
+        }
+
+        public override string NormalizePath([NotNull] string path)
+        {
+            // Need to convert path to application relative (rooted paths are passed in during design time).
+            if (Path.IsPathRooted(path) && path.StartsWith(_applicationRoot, StringComparison.Ordinal))
```

Comment:
There's still no check for a trailing slash here. These checks would pass in a case where `applicationroot = c:\foo` which seems super innocent on the surface. Then you'd end up returning `\bar\baz.cshtml` from this, which is not what you want.